### DISCUSSION TITLE
Building Julia as a standalone

### DISCUSSION THREAD

**Discussion by icould001** [JuliaLang/julia#52751] [General]
Hi,
I am trying to build Julia (v1.9.4) for a RHEL8 system but I have dependency collision between Julia’s dependencies and other packages due to version differences. As a result, decided to build Julia as a standalone (staticly linked) binary using the full source tarball. 
Has anyone tried this before?
Is there an easy way to do this (like setting a flag or something) or do I have to dig into the makefiles (I am not familiar with the details on building Julia, I just have some experience trying to build it as is)?
Thanks for the help!

**Comment by oscardssmith** [comment]
what dependency collision are you seeing? Julia already statically links almost all of it's dependencies

**Reply by icould001** [reply]
It’s statically linked? I got errors from ld complaining about a missing -fPIC so I just assumed everything was dynamically linked.
My main problem was that build time an older version of SuiteSparse was getting linked (libsuitesparseconfig.so.4, the rest of the SuiteSparse libs are the same versions, I think), libuv and libunwind are also older (there are some others, but I don’t remember them all), so I decided to build everything from scratch and not use dynamic libraries at all (as far as possible), but if you say that Julia is already statically linked the I guess I am doing something wrong
Thank you

**Reply by oscardssmith** [reply]
see https://github.com/JuliaLang/julia/blob/master/doc/src/devdocs/build/build.md for the details on what happens by default. So Suitesparse, libuv and libunwind are all downloaded by julia by default (did you set `USE_BINARYBUILDER=0`? If so, I would recommend unsetting that).
